### PR TITLE
Add PKCE to interactive flow

### PR DESCRIPTION
**Pull request by bgavrilMS**
There are 2 commits as part of this: 

- some test refactoring because Lab Api were out of date relative to MSAL (first commit - no need for in-depth review)
- add PKCE support (second commit)

### CODE REVIEW

File: tests/Test.ADAL.NET.Unit.net45/InteractiveAuthTests.cs [hunk 1]
```diff
@@ -169,20 +170,31 @@ public async Task ExtendedLifetimeRetryAsync()
         public async Task AcquireTokenPositiveWithoutUserIdAsync()
         {
             _context = new AuthenticationContext(AdalTestConstants.DefaultAuthorityCommonTenant, new TokenCache());
-            MockHelpers.ConfigureMockWebUI(new AuthorizationResult(AuthorizationStatus.Success,
-                AdalTestConstants.DefaultRedirectUri + "?code=some-code"));
-            AdalHttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler(AdalTestConstants.GetTokenEndpoint(AdalTestConstants.DefaultAuthorityCommonTenant))
+            var mockWebUI = MockHelpers.ConfigureMockWebUI(
```
Comment (bgavrilMS) [inline tests/Test.ADAL.NET.Unit.net45/InteractiveAuthTests.cs:173]: Apart from this unit test, all the Selenium tests now use PKCE. We test AAD, ADFS 2,3 and 4, just like MSAL.

File: tests/Test.ADAL.NET.Unit.net45/InteractiveAuthTests.cs [hunk 2]
```diff
@@ -169,20 +170,31 @@ public async Task ExtendedLifetimeRetryAsync()
...
+                new AuthorizationResult(
+                    AuthorizationStatus.Success,
```
Comment (bgavrilMS) [inline tests/Test.ADAL.NET.Unit.net45/InteractiveAuthTests.cs:175]: There is an extra Integration test that needs to be written - what happens if the PKCE code does not match the code chanllenge. I plan to do this test when I expose the ICustomWebUI interface.

File: tests/Test.ADAL.NET.Common/Mocks/MockHttpMessageHandler.cs [hunk 3]
```diff
@@ -75,15 +77,17 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
...
             {
                 Assert.IsFalse(string.IsNullOrEmpty(uri.Query));
-                IDictionary<string, string> inputQp = EncodingHelper.ParseKeyValueList(uri.Query.Substring(1), '&', true, null);
-                foreach (var key in QueryParams.Keys)
+
+                foreach (var key in QueryParamsToValidate.Keys)
                 {
-                    Assert.IsTrue(inputQp.ContainsKey(key));
-                    Assert.AreEqual(QueryParams[key], inputQp[key]);
+                    Assert.IsTrue(this.ActualQueryOrFormsParams.ContainsKey(key));
```
Comment (MarkZuber) [inline tests/Test.ADAL.NET.Common/Mocks/MockHttpMessageHandler.cs]: nit:  remove "this."

**Review by MarkZuber — APPROVED** [review]
:shipit:

**Pull request merged**